### ISSUE TITLE
Error with lastest version @types/d3

### ISSUE THREAD

**Issue by raulmartinramos** [DefinitelyTyped/DefinitelyTyped#48407]
With the latest version several errors araise.

Property 'event' does not exist on type
Property 'mouse' does not exist on type

**Comment by raulmartinramos** [comment]
d3-selection has been changed from "d3-selection module 1.4" to "d3-selection module 2.0"  and several entries have been
removed like "export function mouse(container: ContainerElement): [number, number];"
Is it right or it is a mistake?

**Comment by morgun** [comment]
We're hit by the same issue.

**Comment by morgun** [comment]
Fixed by manually specifying correct major version for peer dependencies
```
    "@types/d3-array": "1.2.7",
    "@types/d3-axis": "1.0.14",
    "@types/d3-brush": "1.1.4",
    "@types/d3-chord": "1.0.10",
    "@types/d3-collection": "1.0.8",
    "@types/d3-color": "1.4.1",
    "@types/d3-contour": "1.3.1",
    "@types/d3-dispatch": "1.0.9",
    "@types/d3-drag": "1.2.5",
    "@types/d3-dsv": "1.2.1",
    "@types/d3-ease": "1.0.10",
    "@types/d3-fetch": "1.2.2",
    "@types/d3-force": "1.2.2",
    "@types/d3-format": "1.4.1",
    "@types/d3-geo": "1.12.1",
    "@types/d3-hierarchy": "1.1.7",
    "@types/d3-interpolate": "1.4.2",
    "@types/d3-path": "1.0.9",
    "@types/d3-polygon": "1.0.8",
    "@types/d3-quadtree": "1.0.8",
    "@types/d3-random": "1.1.3",
    "@types/d3-scale": "2.2.4",
    "@types/d3-scale-chromatic": "1.5.1",
    "@types/d3-selection": "1.4.3",
    "@types/d3-shape": "1.3.4",
    "@types/d3-time": "1.1.1",
    "@types/d3-time-format": "2.3.1",
    "@types/d3-timer": "1.0.10",
    "@types/d3-transition": "1.3.1",
    "@types/d3-voronoi": "1.1.9",
    "@types/d3-zoom": "1.8.1",
```

**Comment by Ledragon** [comment]
@Methuselah96 could you check this? did you specify in d3 v5 that it depends on d3-selection v1?

**Comment by Methuselah96** [comment]
There are some issues if you install older versions of `@types/d3` (e.g., `@types/d3@5.9.5`). You need to pull the latest version (`@types@5.16.2`) in order to avoid these errors.

This is unfortunate, but there is no way around it. The earlier versions of `@types/d3` did not have all of the packages that it depends on pinned to the correct major version, so it will automatically install the latest version which includes the breaking changes from the new major versions. If you ever have these problems, just pull the latest version of `@types/d3` which should have the correct pinned version numbers.

**Comment by novasilva-admin** [comment]
d3-selection 2.0 is part of d3 v6.0, so why this is added to the Types from version d3 5.x ?

**Comment by Methuselah96** [comment]
@novasilva-admin
Unfortunatley `@types/d3@<=5.7.4` depends on `@types/d3-selection@*` so it will use the latest `@types/d3-selection` which happens to be `@types/d3-selection@2`. There's no way to fix the old versions to be correctly pinned to `@types/d3-selection@^1`.

`@types/d3@>=5.9.0` correctly depends on `@types/d3-selection@^1` so make sure you are using the latest version of `@types/d3` in order to get the correct version of `@types/d3-selection`.

**Comment by spinlud** [comment]
Hi!
I am using `d3@6.5.0` and `@types/d3@6.3.0`, it can't find property `element`:

```ts
import * as d3 from 'd3';

d3.event.transform // <-- TS2339: Property 'event' does not exist on type 'typeof import("/.../node_modules/@types/d3/index")'.
```

**Comment by Methuselah96** [comment]
Can you link to some documentation showing the property you're trying to access?

Also, ideally it would be great if you could create a separate issue for this since it's not related to this issue (or at least a new case that deserves its own issue).
